pod.go

```diff
@@ -562,6 +562,10 @@ func createPod(podConfig PodConfig) (*Pod, error) {
 		return p, nil
 	}
 
+	if err := p.agent.createPod(p); err != nil {
```

Comment:
This new statement makes the function reaching the `gocyclo=15` limit. Please split `createPod(podConfig PodConfig)` so that you fix this issue.